gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/purger/HivePurgerPublisher.java

```diff
@@ -41,12 +79,53 @@ public void publishData(Collection<? extends WorkUnitState> states) {
     for (WorkUnitState state : states) {
       if (state.getWorkingState() == WorkUnitState.WorkingState.SUCCESSFUL) {
         state.setWorkingState(WorkUnitState.WorkingState.COMMITTED);
+        submitEvent(state, ComplianceEvents.Purger.WORKUNIT_COMMITTED);
       } else {
         state.setWorkingState(WorkUnitState.WorkingState.FAILED);
+        submitEvent(state, ComplianceEvents.Purger.WORKUNIT_FAILED);
       }
     }
   }
 
+  private void submitEvent(WorkUnitState state, String name) {
+    WorkUnit workUnit = state.getWorkunit();
+    Map<String, String> metadata = new HashMap<>();
+    metadata.put(ComplianceConfigurationKeys.WORKUNIT_RECORDSREAD, state.getProp(ComplianceConfigurationKeys.NUM_ROWS));
+    metadata.put(ComplianceConfigurationKeys.WORKUNIT_BYTESREAD,
+        getDataSize(workUnit.getProp(ComplianceConfigurationKeys.RAW_DATA_SIZE),
+            workUnit.getProp(ComplianceConfigurationKeys.TOTAL_DATA_SIZE)));
+    Optional<HivePartitionDataset> dataset =
+        DatasetUtils.findDataset(workUnit.getProp(ComplianceConfigurationKeys.PARTITION_NAME), this.datasets);
+    if (!dataset.isPresent()) {
+      return;
+    }
+
+    HivePartitionDataset hivePartitionDataset = dataset.get();
+
+    metadata.put(ComplianceConfigurationKeys.WORKUNIT_RECORDSWRITTEN, DatasetUtils
+        .getPropertyFromDataset(hivePartitionDataset, ComplianceConfigurationKeys.NUM_ROWS,
+            ComplianceConfigurationKeys.DEFAULT_NUM_ROWS));
+    metadata.put(ComplianceConfigurationKeys.WORKUNIT_BYTESWRITTEN, getDataSize(DatasetUtils
+        .getPropertyFromDataset(hivePartitionDataset, ComplianceConfigurationKeys.RAW_DATA_SIZE,
+            ComplianceConfigurationKeys.DEFAULT_RAW_DATA_SIZE), DatasetUtils
+        .getPropertyFromDataset(hivePartitionDataset, ComplianceConfigurationKeys.TOTAL_DATA_SIZE,
+            ComplianceConfigurationKeys.DEFAULT_TOTAL_DATA_SIZE)));
+
+    metadata.put(ComplianceConfigurationKeys.PARTITION_NAME, hivePartitionDataset.datasetURN());
+
+    this.eventSubmitter.submit(name, metadata);
+  }
+
+  private String getDataSize(String rawDataSizeString, String totalDataSizeString) {
```

Comment:
Could you drop 'String' from the parameter names so that it reads rawDataSize, and totalDataSize?